### ISSUE TITLE
nginx ingress (1.10) controller not working on k8s 1.18.14 

### ISSUE THREAD

**Issue by ItsDevender** [nginxinc/kubernetes-ingress#1331]
**Describe the bug**

nginx controller not resolving to the ingress resource service name and giving 500 in the controller log. attached is the application ingress manifest for your reference. 52.x.x.x is the controller loadbalancer serviceIP that i have used here (using .nip.io).

Though app-service-name is working fine when used but nginx ingress controller is not resolving from its service-IP (52.x.x.x).

I have deployed the nginx ingress as per documentation (using Manifest as well as helm3) no luck: https://docs.nginx.com/nginx-ingress-controller/installation/installation-with-manifests/

**To Reproduce**
1. install nginx ingress 1.10 
2. expose controller by creating a service (load balancer) 
3. take that serviceIP and create your application ingress resource to route request through controller
3. use the ingress and it throws 500 in the log. 
4. Though google dig (https://toolbox.googleapps.com/apps/dig/) does resolves to the controller IP  (as per nip.io) but it does NOT resolve further to servicename and hence throwing 500 in controller log

**Expected behavior**

application ingress that was created shall route the request from controller to the application servicename when someone uses its (curl or http). 

**Your environment**
* Version of the Ingress Controller - 1.10
* Version of Kubernetes - 1.18.x
* Kubernetes platform - Azure (AKS)
* Using NGINX or NGINX Plus - NGINX

**Additional context**
 
Last November (2020) I have installed 1.8.1 in couple of k8s clusters (1.16/17), deployment was easy - no post deployment issue or inconsistency. But Recently in one of those cluster/env., one of the app-service that is used as ingress resource (as backend) got changed (deleted and redeployed by developers) - and after that ingress was throwing 500 and not resolving to the service name. I have upgraded the k8s api and also the nginx ingress to 1.10 - but no luck. 

Please advise what is the reason of controller not resolving and identifying the app-service that is defined in the ingress.
[app-service.txt](https://github.com/nginxinc/kubernetes-ingress/files/5913960/app-service.txt)
[app-ingress.txt](https://github.com/nginxinc/kubernetes-ingress/files/5913961/app-ingress.txt)

**Comment by pleshakov** [comment]
Hi @ItsDevender 

Could you try to connect to http://app-52.172.156.118.nip.io again and share [the logs](https://docs.nginx.com/nginx-ingress-controller/troubleshooting/#checking-the-ingress-controller-logs) that correspond to your request? Also, could check if there are any errors/warnings in the logs? This will help us understand where the 500 error is coming from.

**Comment by ItsDevender** [comment]
Thanks @pleshakov . Following are the details:

on the botton you can see the "500" in controller log. on top you can see the backend service name - its internal IP/External IP, description of the ingress - and you can see its showing the backend pods IPs getting resolved under its service. 

Somehow controller is not able to reach any of the pod-ip (using the service name).

Let me know if you need anything more or want me try anything to do and share...

![details_issue](https://user-images.githubusercontent.com/32868104/106735243-fe447380-6639-11eb-87a1-67436a73d124.JPG)

**Comment by pleshakov** [comment]
thanks for providing the details.

from looking into the logs, 500 error is coming from the backend pod, not NGINX. Could you double check the backend logs for any errors there?

**Comment by ItsDevender** [comment]
@pleshakov  - no there is no any such error in backend. Dont know where the request has gone - it came to controller and then it got muted - not hitting to svc end point (viaduct-v4-fhir) - it does have it internal/external end point based on which its resolving to the app-pods.

**Comment by pleshakov** [comment]
@ItsDevender 
I suggest extending the log format via [ConfigMap](https://docs.nginx.com/nginx-ingress-controller/configuration/global-configuration/configmap-resource/#logging) so that it includes the name of the upstream server (backend pod):

```yaml
kind: ConfigMap
apiVersion: v1
metadata:
  name: nginx-config
  namespace: nginx-ingress
data:
  log-format: '$remote_addr - $remote_user [$time_local] "$request" $status $body_bytes_sent "$http_referer" "$http_user_agent" "$http_x_forwarded_for" ua="$upstream_addr"'
  ```
  
 And send a request again. In the logs you should be able to see the IP address and the port of the backend pod. For example:
 ```
 127.0.0.1 - - [03/Feb/2021:23:35:19 +0000] "GET /coffee HTTP/1.1" 200 156 "-" "curl/7.54.0" "-" ua="10.0.1.30:80"
 ```

**Comment by ItsDevender** [comment]
@pleshakov thank you - i will do that and share the details soon

**Comment by ItsDevender** [comment]
sorry for delay @pleshakov . I did put the line to have details of what upstream server (backend pod) are coming in log when i sent request to the controller. I do see in the logs that the pod internal IPs were coming but still controller is giving 500. Really dont know what is making it 500. pods are running fine and they do respond - as app-service is able to respond to the request when i send to it.  both the backend are served under the same app-service. So i dont understand why same pods are responding to the app-service but not nginx controller. 

To check further - I deployed traefik proxy (v1.7) on the cluster and that is also working - I have even upgraded the cluster to 1.18.x and deployed the nginx ingress controller v1.10 (both via manifest and helm3 - but in both cases same result - 500 with backend pod ips in log).

for refernce:
![nginx-controller-issue](https://user-images.githubusercontent.com/32868104/107418007-26e9d300-6b3c-11eb-96b4-83e21939b147.JPG)

Actually one of our app balances the load well using least-conn lb_method and that is why we were preferring nginx - I dont know how traefik (wrr - weighted round robin) will work (still need to test). 

Until we are able to deploy the nginx ingress successfully, we will continue to use traefik to check if its better alternative till we are able to work with nginx-ingress 

Thanks 
-Devender Singh
